src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/TypeConverter.cs

```diff
@@ -232,6 +233,8 @@ protected Exception GetConvertToException(object value, Type destinationType)
         /// the specified context.
         ///
         /// </summary>
+        [RequiresUnreferencedCode("The Type of value cannot be statically discovered.")]
+        [DynamicDependency(DynamicallyAccessedMemberTypes.PublicParameterlessConstructor | DynamicallyAccessedMemberTypes.PublicFields, typeof(BrowsableAttribute))]
```

Comment:
just for clarification: is the idea of having a dynamicDependency attribute here just to do best effort of preserving what is needed? I ask since it also has RequiresUnreferencedCode so we know it won't be safe anyway.

Reply:
This method is doing 2 things that are "unsafe".

1. It is filtering attributes
2. It is calling `GetProperties` passing in an object, which eventually gets `GetType()` called on it.

We can make the first thing "safe" by applying the DynamicDependency. So calling this method will preserve the correct things on the `BrowsableAttribute` class. But the 2nd thing can't be made safe, so it is marked with `RUC`. Consumers of this method who wish to make it safe, can manually mark `value`'s Type as needed, but they would need to dig into the code to figure out why `BrowsableAttribute` is broken if we didn't use `DynamicDependency`.

So in short, this makes it as "safe" as we can make it.